client/state/happychat/actions.js

```diff
@@ -86,6 +87,9 @@ export const requestTranscript = () => ( dispatch, getState ) => {
 export const connectChat = () => ( dispatch, getState ) => {
 	const state = getState();
 	const user = getCurrentUser( state );
+	const locale = getCurrentUserLocale( state );
```

Comment:
Just a heads up that this may return `null`, though I'm guessing the happychat service will fall back to the default locale if that's the case considering it's `undefined` currently.